### PR TITLE
DCA Judge Screen Change

### PR DESCRIPTION
**Pull request by nitish-beehyv**
finished dca pending task reject and approve from home page and hearing page,
added infobox and integrate modal and popup,
feature #2745 and #2746

## Release Notes

- **New Features**
	- Enhanced UI for delay applications with new notification chips in multiple components.
	- Conditional rendering based on the status of delay applications and user roles.
	- Integration of a modal for managing pending tasks related to delay condonation.

- **Bug Fixes**
	- Improved error handling in delay condonation application processes.

- **Documentation**
	- Updated configurations to include new task types for delay condonation submissions.

- **Refactor**
	- Streamlined user information handling and state management across components.

### CODE REVIEW

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/EvidenceHeader.js [hunk 1]
```diff
@@ -5,12 +5,42 @@ import React, { useMemo, useState } from "react";
 import { useTranslation } from "react-i18next";
 import { VideoIcon } from "./CustomSVGs";
 
-const EvidenceHearingHeader = ({ hearing, caseData, filingNumber, setActiveTab, activeTab, onAddParty, hearingLink }) => {
+const delayCondonationStylsMain = {
+  padding: "6px 8px",
+  borderRadius: "999px",
+  backgroundColor: "#E9A7AA",
+  width: "fit-content",
+};
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/EvidenceHeader.js:13]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/EvidenceHeader.js [hunk 2]
```diff
@@ -5,12 +5,42 @@
...
+
+const delayCondonationTextStyle = {
+  margin: "0px",
+  fontFamily: "Roboto",
+  fontSize: "14px",
+  fontWeight: 400,
+  lineHeight: "16.41px",
+  textAlign: "center",
+  color: "#231F20",
+};
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/EvidenceHeader.js:23]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/EvidenceHeader.js [hunk 3]
```diff
@@ -129,6 +161,15 @@
...
       </div>
+      {delayCondonationData?.delayCondonationType?.code === "NO" && isJudge && (
+        <div className="delay-condonation-chip" style={delayCondonationStylsMain}>
+          <p style={delayCondonationTextStyle}>
+            {delayCondonationData?.delayCondonationType?.isDcaSkippedInEFiling || isDelayApplicationPending
+              ? t("DELAY_CONDONATION_FILED")
+              : t("DELAY_CONDONATION_NOT_FILED")}
+          </p>
+        </div>
+      )}
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/EvidenceHeader.js:172]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/EndHearing.js [hunk 4]
```diff
@@ -41,7 +42,16 @@
...
+const EndHearing = ({
+  handleEndHearingModal,
+  hearingId,
+  updateTranscript,
+  hearing,
+  transcriptText,
+  disableTextArea,
+  setTranscriptText,
+  isItemPending,
+}) => {
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/EndHearing.js:54]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/EndHearing.js [hunk 5]
```diff
@@ -8,9 +8,10 @@ import { hearingService } from "../../hooks/services";
 
 const Heading = (props) => {
   return (
-    <div style={{ width: "440px", height: "56px" }}>
+    <div style={{ width: "440px" }}>
       <p
         style={{
+          marginBottom: "0px",
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/EndHearing.js:14]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/home/src/components/PendingTaskAccordion.js [hunk 6]
```diff
@@ -38,6 +46,46 @@
...
   };
+  const delayCondonationTextStyle = {
+    margin: "0px",
+    fontFamily: "Roboto",
+    fontSize: "16px",
+    fontWeight: 400,
+    lineHeight: "21.6px",
+    color: "#231F20",
+  };
+
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/home/src/components/PendingTaskAccordion.js:57]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/home/src/components/PendingTaskAccordion.js [hunk 7]
```diff
@@ -38,6 +46,46 @@
...
+            });
+          },
+          actionCancelLabel: "BACK",
+          actionCancelOnSubmit: () => {
+            setIsOpenDCA(false);
+          },
+        },
+      ],
+    };
+  }, [redirectUrl]);
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/home/src/components/PendingTaskAccordion.js:88]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/home/src/components/TaskComponent.js [hunk 8]
```diff
@@ -562,6 +562,7 @@ const TasksComponent = ({
                     <div className="task-section">
                       <PendingTaskAccordion
                         pendingTasks={pendingTaskDataInWeek}
+                        allPendingTasks={[...pendingTaskDataInWeek, ...allOtherPendingTask]}
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/home/src/components/TaskComponent.js:565]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/home/src/components/TaskComponent.js [hunk 9]
```diff
@@ -574,6 +575,7 @@
                     <div className="task-section">
                       <PendingTaskAccordion
                         pendingTasks={allOtherPendingTask}
+                        allPendingTasks={[...pendingTaskDataInWeek, ...allOtherPendingTask]}
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/home/src/components/TaskComponent.js:578]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/home/src/components/PendingTaskAccordion.js [hunk 10]
```diff
@@ -90,12 +138,25 @@
...
+                      );
+                      setIsOpenDCA(true);
+                    } else {
+                      history.push(
+                        `/${window.contextPath}/employee/dristi/home/view-case?caseId=${caseId}&filingNumber=${filingNumber}&tab=Overview`,
+                        {
+                          triggerAdmitCase: true,
+                        }
+                      );
+                    }
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/home/src/components/PendingTaskAccordion.js:159]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/EndHearing.js [hunk 11]
```diff
@@ -91,32 +99,37 @@
...
           }}
+          popupModuleMianStyles={{ paddingLeft: "0px" }}
           popupModuleActionBarStyles={{
             display: "flex",
             justifyContent: "flex-end",
-            position: "absolute",
-            right: 0,
-            bottom: 0,
-            width: "100%",
+            width: "calc(100% + 48px)",
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/EndHearing.js:106]: checked

**Review by TusharKatiyar2002 — APPROVED** [review]

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/EndHearing.js [hunk 12]
```diff
@@ -91,32 +99,37 @@
...
             borderTop: "1px solid #dbd7d2",
             padding: "10px 16px 16px 0px",
+            position: "relative",
+            left: "-24px",
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/EndHearing.js:110]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js [hunk 13]
```diff
@@ -236,14 +255,21 @@
...
-    [nextActions]
-  );
+  const primaryAction = useMemo(() => {
+    const action = casePrimaryActions?.find((action) => nextActions?.some((data) => data.action === action?.action)) || { action: "", label: "" };
+    if (isDelayApplicationPending && action.action === "ADMIT") {
+      action.label = "CS_ADMIT_APPROVE_DCA";
+    }
+    return action;
+  }, [nextActions, isDelayApplicationPending]);
+
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js:265]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/admission/CaseFileAdmission.js [hunk 14]
```diff
@@ -756,7 +773,11 @@ function CaseFileAdmission({ t, path }) {
         },
       },
     };
-    return await DRISTIService.createApplication(applicationReqBody, { tenantId });
+    try {
+      return await DRISTIService.createApplication(applicationReqBody, { tenantId });
+    } catch (error) {
+      console.error("Failed to create applications :>> ", error);
+    }
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/admission/CaseFileAdmission.js:780]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/admission/CaseFileAdmission.js [hunk 15]
```diff
@@ -41,6 +41,22 @@
...
+
+const delayCondonationTextStyle = {
+  margin: "0px",
+  fontFamily: "Roboto",
+  fontSize: "14px",
+  fontWeight: 400,
+  lineHeight: "16.41px",
+  textAlign: "center",
+  color: "#231F20",
+};
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/admission/CaseFileAdmission.js:58]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/home/src/components/PendingTaskAccordion.js [hunk 16]
```diff
@@ -18,10 +19,17 @@ function PendingTaskAccordion({
...
+  allPendingTasks,
 }) {
   const history = useHistory();
   const [isOpen, setIsOpen] = useState(isAccordionOpen);
   const [check, setCheck] = useState(false);
+  const [isOpenDCA, setIsOpenDCA] = useState(false);
+  const [redirectUrl, setRedirectUrl] = useState("");
+
+  const roles = useMemo(() => Digit.UserService.getUser()?.info?.roles?.map((role) => role?.code) || [], []);
+  const isJudge = roles.includes("JUDGE_ROLE");
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/home/src/components/PendingTaskAccordion.js:31]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js [hunk 17]
```diff
@@ -1955,6 +1979,15 @@
...
             <div className="sub-details-text">Code: {caseDetails?.accessCode}</div>
+            {delayCondonationData?.delayCondonationType?.code === "NO" && isJudge && (
+              <div className="delay-condonation-chip" style={delayCondonationStylsMain}>
+                <p style={delayCondonationTextStyle}>
+                  {delayCondonationData?.delayCondonationType?.isDcaSkippedInEFiling || isDelayApplicationPending
+                    ? t("DELAY_CONDONATION_FILED")
+                    : t("DELAY_CONDONATION_NOT_FILED")}
+                </p>
+              </div>
+            )}
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js:1990]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js [hunk 18]
```diff
@@ -1449,10 +1473,10 @@
...
         break;
       case "ADMIT":
-        if (isDelayApplicationPending) {
-          setShowPendingDelayApplication(true);
-          break;
-        }
+        // if (isDelayApplicationPending) {
+        //   setShowPendingDelayApplication(true);
+        //   break;
+        // }
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js:1479]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js [hunk 19]
```diff
@@ -236,14 +255,21 @@
...
+    return action;
+  }, [nextActions, isDelayApplicationPending]);
+
+  const secondaryAction = useMemo(() => {
+    const action = caseSecondaryActions?.find((action) => nextActions?.some((data) => data.action === action?.action)) || { action: "", label: "" };
+    if (isDelayApplicationPending && action.action === "REJECT") {
+      action.label = "CS_REJECT_APPROVE_DCA";
+    }
+    return action;
+  }, [nextActions, isDelayApplicationPending]);
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js:272]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/EndHearing.js [hunk 20]
```diff
@@ -91,32 +99,37 @@
           headerBarMainStyle={{
             padding: "0px 24px 12px 24px",
           }}
+          popupModuleMianStyles={{ paddingLeft: "0px" }}
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/EndHearing.js:102]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/InsideHearingMainPage.js [hunk 21]
```diff
@@ -121,6 +127,35 @@ const InsideHearingMainPage = () => {
...
+        filingNumber,
+        tenantId,
+      },
+      tenantId,
+    },
+    {},
+    filingNumber + "allApplications",
+    filingNumber
+  );
+
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/InsideHearingMainPage.js:142]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/InsideHearingMainPage.js [hunk 22]
```diff
@@ -319,6 +354,8 @@
 
   const attendanceCount = useMemo(() => hearing?.attendees?.filter((attendee) => attendee.wasPresent)?.length || 0, [hearing]);
   const [isRecording, setIsRecording] = useState(false);
+  const [taskType, setTaskType] = useState({});
+
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/hearings/src/pages/employee/InsideHearingMainPage.js:358]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js [hunk 23]
```diff
@@ -46,6 +46,22 @@ const stateSla = {
   NOTICE: 3 * 24 * 3600 * 1000,
 };
 
+const delayCondonationStylsMain = {
+  padding: "6px 8px",
+  borderRadius: "999px",
+  backgroundColor: "#E9A7AA",
+};
+
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js:54]: checked

File: frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js [hunk 24]
```diff
@@ -46,6 +46,22 @@
...
+const delayCondonationTextStyle = {
+  margin: "0px",
+  fontFamily: "Roboto",
+  fontSize: "14px",
+  fontWeight: 400,
+  lineHeight: "16.41px",
+  textAlign: "center",
+  color: "#231F20",
+};
```
[the remark this answers is not in the document]
Comment (nitish-beehyv) [reply frontend/micro-ui/web/micro-ui-internals/packages/modules/dristi/src/pages/employee/AdmittedCases/AdmittedCase.js:63]: checked

**Pull request merged**